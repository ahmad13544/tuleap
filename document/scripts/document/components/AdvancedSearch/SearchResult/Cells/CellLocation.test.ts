/**
 * Copyright (c) Enalean, 2022 - present. All Rights Reserved.
 *
 * This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 */

import type { ItemSearchResult } from "../../../../type";
import { shallowMount } from "@vue/test-utils";
import localVue from "../../../../helpers/local-vue";
import CellLocation from "./CellLocation.vue";
import VueRouter from "vue-router";

describe("CellLocation", () => {
    it("should display path via the router link and the folder separators", () => {
        const router = new VueRouter({
            routes: [
                {
                    path: "/",
                    name: "search",
                },
            ],
        });
        const wrapper = shallowMount(CellLocation, {
            localVue,
            propsData: {
                item: {
                    parents: [
                        {
                            id: 120,
                            title: "Path",
                        },
                        {
                            id: 121,
                            title: "To",
                        },
                        {
                            id: 122,
                            title: "Folder",
                        },
                    ],
                } as unknown as ItemSearchResult,
            },
            router,
        });

        expect(wrapper.element).toMatchSnapshot();
    });
});
