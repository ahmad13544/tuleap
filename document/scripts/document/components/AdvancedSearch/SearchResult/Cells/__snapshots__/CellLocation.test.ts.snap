// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CellLocation should display path via the router link and the folder separators 1`] = `
<td>
  <router-link-stub
    ariacurrentvalue="page"
    custom="true"
    data-test="document-cell-location-parent-item-title"
    event="click"
    tag="a"
    to="[object Object]"
  />
   
  <span>
    
            /
            
  </span>
  <router-link-stub
    ariacurrentvalue="page"
    custom="true"
    data-test="document-cell-location-parent-item-title"
    event="click"
    tag="a"
    to="[object Object]"
  />
   
  <span>
    
            /
            
  </span>
  <router-link-stub
    ariacurrentvalue="page"
    custom="true"
    data-test="document-cell-location-parent-item-title"
    event="click"
    tag="a"
    to="[object Object]"
  />
   
  <!---->
</td>
`;
